translate/storage/pypo.py

```diff
@@ -989,9 +990,10 @@ def addcomment(thepo):
                     origpo = id_dict[id]
                     if origpo not in markedpos and id:
                         # if it doesn't have an id, we already added msgctxt
```

Comment:
The code was apparently written with the assumption that `id` will be blank/None if the `msgctxt` was already handled. If this is not the case, the `id` should be removed from the condition and the comment removed as well. Maybe you can also merge the `origpo.msgctxt` condition here?

Reply:
I think you are right. I've applied your suggestion.